ScoutSuite/providers/aws/configs/regions_config.py

```diff
@@ -15,5 +15,28 @@ def __init__(self, service):
         self['regions'] = {}
         for region in await facade.build_region_list(self._service, chosen_regions, partition_name):
             self['regions'][region] = {}
-            
+
         self['regions_count'] = len(self['regions'])
+
+
+class ScopedResources(Resources):
+    def __init__(self, key):
+        self.key = key
+
+    async def fetch_all(self, scope):
+        resources = {}
+        for raw_resource in await self.get_resources_in_scope(scope):
+            name, ressource = self.parse_resource(raw_resource)
+            resources[name] = ressource
+
+        self[self.key + '_count'] = len(resources)
+        self[self.key] = resources
```

Comment:
Not sure this is the best solution. Might also just do `self[name] = resource`, and have the class using a `ScopedResources` handle the count/dic name.